downloader.py

```diff
@@ -0,0 +1,310 @@
+        return
+      else:
+        log("Uncrecognized option, aborting...")
+        sys.exit(0)
+    
+    try:
+      download_command = self.create_download_command()
+      subprocess.run(download_command,
+                              shell=True,
+                              stderr=subprocess.STDOUT)
+    except Exception as e:
+      if debug:
+        traceback.print_exc()
+
+      log("ERROR downloading dataset. Skipping... Please use --debug to show error")
+      return
+
+    log("Download success! Going to unpack archive")
+    log("*** UNPACKING ***")
+    log("Please wait while archive is being unpacked. You won't see any progress until unpack is done because it is done in silent mode, not to clutter output with list of files")
+
+    unpack_command = self.create_unpack_command()
+
+    try:
+      subprocess.run(unpack_command,
+                     shell=True,
+                     stderr=subprocess.STDOUT)     
+      
+      if self.check_for_unpacking_folder:
+        if not Path(directory_path).exists():
+          raise Exception(
+            f"""It seems there is a problem with unpacking!
+Folder with inner files not created after unpacking!
+There should be folder: {directory_path} after unpacking, but it's not there!
+Please check it for yourself"""
+          )
+
+      log("Unpack successfull!")
+
+      log("Removing archive file...")
+
+      Path(filepath).unlink()
+
+      log("Archive file removed!")
+    except Exception as e:
+      if debug:
+        traceback.print_exc()
+      log("ERROR unpacking! Skipping... Please use --debug to show error")
+      return
+
+if test_run:
+  download_requests = [
+    DownloadRequest("zipTest", "https://github.com/marmelroy/Zip/blob/master/examples/Sample/Sample/Images.zip?raw=true", download_path, check_for_unpacking_folder=False)
+  ]
+else:
+  download_requests = [
+    DownloadRequest("annotations_trainval2017", "http://images.cocodataset.org/annotations/stuff_annotations_trainval2017.zip", download_path),
+    DownloadRequest("val2017", "http://images.cocodataset.org/zips/val2017.zip", download_path),
+    DownloadRequest("train2017", "http://images.cocodataset.org/zips/train2017.zip", download_path),
+  ]
+
+log("Checking path for downloading...")
+
+if not Path(download_path).exists():
+  log(f"Path {download_path} does not exist! Creating it...")
+
+  try:
+    Path(download_path).mkdir(parents=True)
+  except Exception as e:
+    if debug:
+      traceback.print_exc()
+    log("Critical error creating path! Aborting. (more info with --debug option)")
+    sys.exit(0)
+else:
+  log("Path seems to be ok!")
+
+log("Going to download needed datasets!")
+sleep(1)
+
+for dr in download_requests:
+  log("")
+  log("")
+  log(f"Dataset [{dr.name}] with size: {dr.remote_filesize}")
+  log("*** DOWNLOADING ***")
+  
+  dr.download()
+  
+log("")
+log("")
+log("*** SUCCESS ***")
+log("All datasets downloads complete!")
+log("*** SUCCESS ***")
+sys.exit(0)
```

Comment:
@idchlife 
I think this line doesn't need because python call `exit(0)` when meet the end of the script.

Reply:
@tucan9389 I agree! Removed this line